The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/11802437329/job/32878113995) [(plain)](https://github.com/rust-lang/rust/commit/71786ee619cf451599ac28e608393e214d9c2148/checks/32878113995/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain

COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---
#13 2.856 Building wheels for collected packages: reuse
#13 2.857   Building wheel for reuse (pyproject.toml): started
#13 3.106   Building wheel for reuse (pyproject.toml): finished with status 'done'
#13 3.108   Created wheel for reuse: filename=reuse-4.0.3-cp310-cp310-manylinux_2_35_x86_64.whl size=132720 sha256=026f3bb0f1aa8090b861fd0a0939cb1a782396d84c8aab7875096557d637a0f6
#13 3.108   Stored in directory: /tmp/pip-ephem-wheel-cache-y8kmaxw1/wheels/3d/8d/0a/e0fc6aba4494b28a967ab5eaf951c121d9c677958714e34532
#13 3.111 Installing collected packages: boolean-py, binaryornot, tomlkit, reuse, python-debian, markupsafe, license-expression, jinja2, chardet, attrs
#13 3.507 Successfully installed attrs-23.2.0 binaryornot-0.4.4 boolean-py-4.0 chardet-5.2.0 jinja2-3.1.4 license-expression-30.3.0 markupsafe-2.1.5 python-debian-0.1.49 reuse-4.0.3 tomlkit-0.13.0
#13 3.507 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#13 4.061 Collecting virtualenv
#13 4.061 Collecting virtualenv
#13 4.097   Downloading virtualenv-20.27.1-py3-none-any.whl (3.1 MB)
#13 4.151      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.1/3.1 MB 59.7 MB/s eta 0:00:00
#13 4.190 Collecting distlib<1,>=0.3.7
#13 4.193   Downloading distlib-0.3.9-py2.py3-none-any.whl (468 kB)
#13 4.201      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 469.0/469.0 KB 86.1 MB/s eta 0:00:00
#13 4.241 Collecting filelock<4,>=3.12.2
#13 4.245   Downloading filelock-3.16.1-py3-none-any.whl (16 kB)
#13 4.280 Collecting platformdirs<5,>=3.9.1
#13 4.283   Downloading platformdirs-4.3.6-py3-none-any.whl (18 kB)
#13 4.367 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#13 4.546 Successfully installed distlib-0.3.9 filelock-3.16.1 platformdirs-4.3.6 virtualenv-20.27.1
#13 DONE 4.6s

#14 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#14 DONE 0.0s
---
DirectMap4k:      319424 kB
DirectMap2M:     8069120 kB
DirectMap1G:    10485760 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
    Finished `dev` profile [unoptimized] target(s) in 0.04s
##[endgroup]
downloading https://static.rust-lang.org/dist/2024-10-16/rustfmt-nightly-x86_64-unknown-linux-gnu.tar.xz
extracting /checkout/obj/build/cache/2024-10-16/rustfmt-nightly-x86_64-unknown-linux-gnu.tar.xz to /checkout/obj/build/x86_64-unknown-linux-gnu/rustfmt
---
Diff in /checkout/compiler/rustc_ast/src/visit.rs:820:
     V::Result::output()
 }
 
-pub fn walk_fn_header<'a, V: Visitor<'a>>(
-    visitor: &mut V,
-    fn_header: &'a FnHeader
-) -> V::Result {
+pub fn walk_fn_header<'a, V: Visitor<'a>>(visitor: &mut V, fn_header: &'a FnHeader) -> V::Result {
     let FnHeader { safety: _, coroutine_kind, constness: _, ext: _ } = fn_header;
     visit_opt!(visitor, visit_coroutine_kind, coroutine_kind.as_ref());
     V::Result::output()
fmt error: Running `"/checkout/obj/build/x86_64-unknown-linux-gnu/rustfmt/bin/rustfmt" "--config-path" "/checkout" "--edition" "2021" "--unstable-features" "--skip-children" "--check" "/checkout/compiler/rustc_llvm/src/lib.rs" "/checkout/compiler/rustc_ast_pretty/src/pprust/state.rs" "/checkout/compiler/rustc_ast_pretty/src/pprust/state/item.rs" "/checkout/compiler/rustc_ast_pretty/src/pprust/state/fixup.rs" "/checkout/compiler/rustc_ast_pretty/src/pprust/state/expr.rs" "/checkout/compiler/rustc_ast_pretty/src/pprust/tests.rs" "/checkout/compiler/rustc_ast_pretty/src/pprust/mod.rs" "/checkout/compiler/rustc_ast_pretty/src/pp/convenience.rs" "/checkout/compiler/rustc_ast_pretty/src/pp/ring.rs" "/checkout/compiler/rustc_ast_pretty/src/pp.rs" "/checkout/compiler/rustc_ast_pretty/src/helpers.rs" "/checkout/compiler/rustc_ast_pretty/src/lib.rs" "/checkout/compiler/rustc_errors/src/annotate_snippet_emitter_writer.rs" "/checkout/compiler/rustc_errors/src/registry.rs" "/checkout/compiler/rustc_errors/src/emitter.rs" "/checkout/compiler/rustc_errors/src/markdown/tests/term.rs" "/checkout/compiler/rustc_errors/src/markdown/tests/parse.rs" "/checkout/compiler/rustc_errors/src/markdown/term.rs" "/checkout/compiler/rustc_errors/src/markdown/parse.rs" "/checkout/compiler/rustc_errors/src/markdown/mod.rs" "/checkout/compiler/rustc_errors/src/error.rs" "/checkout/compiler/rustc_errors/src/snippet.rs" "/checkout/compiler/rustc_errors/src/lock.rs" "/checkout/compiler/rustc_errors/src/diagnostic.rs" "/checkout/compiler/rustc_errors/src/codes.rs" "/checkout/compiler/rustc_errors/src/tests.rs" "/checkout/compiler/rustc_errors/src/diagnostic_impls.rs" "/checkout/compiler/rustc_errors/src/json.rs" "/checkout/compiler/rustc_errors/src/lib.rs" "/checkout/compiler/rustc_errors/src/styled_buffer.rs" "/checkout/compiler/rustc_errors/src/json/tests.rs" "/checkout/compiler/rustc_errors/src/translation.rs" "/checkout/compiler/rustc_ast/src/node_id.rs" "/checkout/compiler/rustc_ast/src/attr/mod.rs" "/checkout/compiler/rustc_ast/src/token.rs" "/checkout/compiler/rustc_ast/src/ast_traits.rs" "/checkout/compiler/rustc_ast/src/visit.rs" "/checkout/compiler/rustc_ast/src/ptr.rs" "/checkout/compiler/rustc_ast/src/expand/autodiff_attrs.rs" "/checkout/compiler/rustc_ast/src/expand/typetree.rs" "/checkout/compiler/rustc_ast/src/expand/allocator.rs" "/checkout/compiler/rustc_ast/src/expand/mod.rs" "/checkout/compiler/rustc_ast/src/mut_visit.rs" "/checkout/compiler/rustc_ast/src/tokenstream.rs" "/checkout/compiler/rustc_ast/src/format.rs" "/checkout/compiler/rustc_ast/src/ast.rs" "/checkout/compiler/rustc_ast/src/entry.rs" "/checkout/compiler/rustc_ast/src/util/parser.rs" "/checkout/compiler/rustc_ast/src/util/comments/tests.rs" "/checkout/compiler/rustc_ast/src/util/classify.rs" "/checkout/compiler/rustc_ast/src/util/comments.rs" "/checkout/compiler/rustc_ast/src/util/case.rs" "/checkout/compiler/rustc_ast/src/util/literal.rs" "/checkout/compiler/rustc_ast/src/util/unicode.rs" "/checkout/compiler/rustc_ast/src/lib.rs" "/checkout/compiler/rustc_hir_pretty/src/lib.rs" "/checkout/compiler/rustc_hir_analysis/src/impl_wf_check/min_specialization.rs" "/checkout/compiler/rustc_hir_analysis/src/hir_wf_check.rs" "/checkout/compiler/rustc_hir_analysis/src/collect/predicates_of.rs" "/checkout/compiler/rustc_hir_analysis/src/collect/resolve_bound_vars.rs" "/checkout/compiler/rustc_hir_analysis/src/collect/item_bounds.rs" "/checkout/compiler/rustc_hir_analysis/src/collect/type_of.rs" "/checkout/compiler/rustc_hir_analysis/src/collect/type_of/opaque.rs" "/checkout/compiler/rustc_llvm/build.rs"` failed.
If you're running `tidy`, try again with `--bless`. Or, if you just want to format code, run `./x.py fmt` instead.
  local time: Tue Nov 12 17:35:00 UTC 2024
  network time: Tue, 12 Nov 2024 17:35:00 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/compiler/rustc_ast/src/visit.rs
+++ b/compiler/rustc_ast/src/visit.rs
@@ -820,10 +820,7 @@ pub fn walk_fn_ret_ty<'a, V: Visitor<'a>>(visitor: &mut V, ret_ty: &'a FnRetTy) 
     V::Result::output()
 }
 
-pub fn walk_fn_header<'a, V: Visitor<'a>>(
-    visitor: &mut V,
-    fn_header: &'a FnHeader
-) -> V::Result {
+pub fn walk_fn_header<'a, V: Visitor<'a>>(visitor: &mut V, fn_header: &'a FnHeader) -> V::Result {
     let FnHeader { safety: _, coroutine_kind, constness: _, ext: _ } = fn_header;
     visit_opt!(visitor, visit_coroutine_kind, coroutine_kind.as_ref());
     V::Result::output()

